src/main/java/com/jcabi/log/VerboseProcess.java

```diff
@@ -473,6 +486,87 @@ public Void call() throws Exception {
             }
             return null;
         }
+        
+        private void logFromReader(final BufferedReader reader,
+                final BufferedWriter writer) throws IOException, 
+                ClosedByInterruptException {
+            StringBuilder builder = new StringBuilder();
+            String previous = NULL_STRING;
+            int lineCount = 0;
+            while (true) {
+                if (Thread.interrupted()) {
+                    Logger.debug(
+                        VerboseProcess.class,
+                        "explicitly interrupting read from buffer"
+                    );
+                    break;
+                }
+                if (previous != NULL_STRING) {
+                    builder.append(previous).append(NEW_LINE);
+                    previous = NULL_STRING;
+                }
+                final String line = reader.readLine();
+                if (line == null) {
```

Comment:
@mkordas this is from the existing code, and I can't control how readLine() works